src/fidesctl/api/ctl/database/crud.py

```diff
@@ -110,7 +118,13 @@ async def update_resource(sql_model: Base, resource_dict: Dict) -> Dict:
     with log.contextualize(
         sql_model=sql_model.__name__, fides_key=resource_dict["fides_key"]
     ):
-        await get_resource(sql_model, resource_dict["fides_key"])
+        resource = await get_resource(sql_model, resource_dict["fides_key"])
```

Comment:
I see what you mean here about the difficulty of embedding this check within the function itself, upon further thinking I don't think this is what we need here.

I think maybe adding the check to `routes/crud.py` might be better? That way when we call these functions internally we don't have any restrictions, and our goal is specifically to stop _Users_ from updating the default taxonomy, so I think putting the checks in the routes makes more sense

so `forbid_if_default` might look something like:

```python
if resource_type in protected_types:
    is_default = await get_resource(...)
    if is_default:
        raise ForbiddenError
```

Reply:
okay yeah, makes sense! thanks for your guidance 🙏